ktxspec.adoc

```diff
@@ -871,35 +871,43 @@ A _dfDescriptor_ is useful in the following cases:
 _dfDescriptorBlock_.
 
-* The value of `<<_vkformat,vkFormat>>` and the image data must
-  never contradict the DFD and, vice-versa, the DFD must not
-  contradict `<<_vkformat,vkFormat>>`. Thus the value of `vkFormat`
-  can be used as a shortcut. Only when it is `VK_FORMAT_UNDEFINED`,
-  does an application need to utilitize sample information from the
-  DFD.  (In this case contradiction is impossible.)
-* `colorModel` must be `KHR_DF_MODEL_RGBSDA`
-  or one of the block compressed color models listed in <<KDF13>>
-  Section 5.6 or its successors, currently `KHR_DF_MODEL_BC1A` to
-  `KHR_DF_MODEL_UASTC`.
-* `colorPrimaries` may be any of the available values since
-  conversion of the selected primaries and white point to a display's
-  can be done simply with a 3x3 matrix multiply.
+* If `<<_vkformat,vkFormat>>` is not `VK_FORMAT_UNDEFINED`, DFD's
+  `texelBlockDimension*`, `bytesPlane*`, and sample information fields
+  must match the format's definition. The `colorModel` must be
+  `KHR_DF_MODEL_RGBSDA` or the matching block compressed color model
+  listed in <<KDF13>> Section 5.6 or its successors, currently
+  `KHR_DF_MODEL_BC1A` to `KHR_DF_MODEL_UASTC`.
+* If `<<_vkformat,vkFormat>>` is one of the `+*_SRGB{,_*}+` formats,
+  `transferFunction` must be `KHR_DF_TRANSFER_SRGB`.
+* If `<<_vkformat,vkFormat>>` is not one of the `+*_SRGB{,_*}+` formats
+  and an sRGB variant of that format exists, `transferFunction` should not
+  be `KHR_DF_TRANSFER_SRGB`.
 
 [NOTE]
 ====
-`KHR_DF_PRIMARIES_BT709/SRGB` is recommended for standard dynamic range,
-standard gamut images.
+For example, `VK_FORMAT_R8G8B8A8_UNORM` should not be used with
+`KHR_DF_TRANSFER_SRGB` because there is `VK_FORMAT_R8G8B8A8_SRGB`.
+
+On the other hand, `VK_FORMAT_A2B10G10R10_UNORM_PACK32` may be used with
+`KHR_DF_TRANSFER_SRGB` because there is no sRGB variant of this format.
+====
+
+[NOTE]
+====
+When `<<_vkformat,vkFormat>>` is not one of the `+*_SRGB{,_*}+` formats and the
+transfer function is not linear, the KTX file may be much less portable due to
+limited hardware support of such inputs.
 ====
 
-* `transferFunction` must be `KHR_DF_TRANSFER_LINEAR` or
-  `KHR_DF_TRANSFER_SRGB` since these are the only ones supported by the
-  vast majority of graphics hardware and correct filtering and blending
-  without hardware support is difficult to impossible.
-* If `<<_vkformat,vkFormat>>` is one of the `+*_SRGB{,_*}+` formats
-  then `transferFunction` must be `KHR_DF_TRANSFER_SRGB` and
-  vice-versa.  For all other `vkFormat` values, except
-  `VK_FORMAT_UNDEFINED`, `transfer_function` must be
-  `KHR_DF_TRANSFER_LINEAR`.
+[NOTE]
+====
+`colorPrimaries` may be any of the available values since conversion
+of the selected primaries and white point to a display's can be done
+simply with a 3x3 matrix multiply.
+
+Still, `KHR_DF_PRIMARIES_BT709` / `KHR_DF_PRIMARIES_SRGB` is recommended
+for standard dynamic range, standard gamut images.
+====
```

Comment:
Can we make any recommendation for HDR images? E.g BT.2100.

Reply:
In general, yes. In practice, there are several issues.

- KDFS does not define BT.2100 enum for primaries, although the actual values are the same as in BT.2020, which is defined for DFD.
- 8 bits are not enough to reasonably cover that gamut. All options from 10-bit normalized with PQ encoding to full linear `float32` storage are used in the wild, yet there's no well-accepted standard.
- In some cases, BT.2100 gamut is signalled but not fully utilized (e.g., actual colors stay within P3) for better compatibility.
-  In other cases, DisplayP3 is signalled explicitly and low-precision storage is used, e.g., sRGB curve with 10 bits (or even 8 bits with dithering).

So I don't think there's a simple recommendation that we could put here.

Reply:
Thanks. I'll resolve this then and I'll ask Andrew to add BT.2100 to KDFS. It already has a value for the HLG transfer function which is one of the choices in BT.2100.